__tests__/oas.test.js

```diff
@@ -3,280 +3,362 @@ const { Operation } = require('../src/oas');
+      const operation = new Oas(multipleSecurities).operation('/no-auth', 'post');
+      expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
+    });
 
-  it('should work for multiple securities (&& and ||)', () => {
-    const operation = new Oas(multipleSecurities).operation('/and-or-security', 'post');
+    it('should return empty object if security scheme doesnt exist', () => {
+      const operation = new Oas(multipleSecurities).operation('/unknown-scheme', 'post');
+      expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
+    });
 
-    expect(operation.prepareSecurity().OAuth2).toHaveLength(2);
-    expect(operation.prepareSecurity().Header).toHaveLength(1);
+    it('should return empty if security scheme type doesnt exist', () => {
+      const operation = new Oas(multipleSecurities).operation('/unknown-auth-type', 'post');
+      expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
+    });
   });
 
-  it.todo('should set a `key` property');
+  describe('getHeaders()', () => {
+    it('should return an object containing request headers if params exist', () => {
+      const oas = new Oas(petstore);
+      const uri = `http://petstore.swagger.io/v2/pet/1`;
+      const method = 'DELETE';
 
-  // TODO We dont currently support cookies?
-  it.todo('apiKey/cookie: should return with a type of Cookie');
+      const { logOperation } = oas.findOperation(uri, method);
+      const operation = new Operation(oas, logOperation.url.path, logOperation.url.method, logOperation.operation);
 
-  it.todo('should throw if attempting to use a non-existent scheme');
+      expect(operation.getHeaders()).toMatchObject({
+        request: ['api_key'],
+        response: [],
+      });
+    });
 
-  it('should return empty object if no security', () => {
-    const operation = new Oas(multipleSecurities).operation('/no-auth', 'post');
-    expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
-  });
+    it('should return an object containing request headers if security exists', () => {
+      const oas = new Oas(multipleSecurities);
+      const uri = 'http://example.com/multiple-combo-auths';
+      const method = 'POST';
 
-  it('should return empty object if security scheme doesnt exist', () => {
-    const operation = new Oas(multipleSecurities).operation('/unknown-scheme', 'post');
-    expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
-  });
+      const { logOperation } = oas.findOperation(uri, method);
+      const operation = new Operation(oas, logOperation.url.path, logOperation.url.method, logOperation.operation);
 
-  it('should return empty if security scheme type doesnt exist', () => {
-    const operation = new Oas(multipleSecurities).operation('/unknown-auth-type', 'post');
-    expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
+      expect(operation.getHeaders()).toMatchObject({
+        request: ['apiKey'],
```

Comment:
Based on the security scheme `type` and `scheme`, we should match them up to the header that would end up serving that (eg. `apiKey` could either match up with the `Authentication` header, or it could sometimes be supplied as a query param and shouldn't be returned here).

There's some work on handling the response from `prepareSecurity` in the explorer that might help:

* https://github.com/readmeio/api-explorer/blob/master/packages/api-explorer/src/Response.jsx
* https://github.com/readmeio/api-explorer/blob/master/packages/api-explorer/src/SecurityInput.jsx

The OAS docs on schemes might help as well https://swagger.io/docs/specification/authentication/#securitySchemes

Reply:
Thanks Jon, I'll put some work into this!

Reply:
@erunion I've updated this, please take another look. I'll finish tests for additions before merge.